src/main/scala/com/fulcrumgenomics/commons/io/Io.scala

```diff
@@ -45,10 +47,10 @@ trait IoUtil {
   val StdOut: Path = PathUtil.pathTo("/dev/stdout")
   val DevNull: Path = PathUtil.pathTo("/dev/null")
 
+  /** The level of compression to use when writing compressed output. */
+  var compressionLevel: Int
```

Comment:
Alternatively, in fgbio we could create a setter for `compressionLevel`.  I think this is overkill, as there's no logic that this enables that isn't supported by `var`, for example if we need some custom logic or redirection in the getter or setter.  I thought I'd mention it though.
 
```scala
trait Foo extends Io {
  private var _compressionLevel: Int = super.compressionLevel
  def compressionLevel: Int = this._compressionLevel
  def compressionLevel_=(level: Int): Unit = this._compressionLevel = level
}
```

Reply:
I'm not sure I love having the `var` in the trait if I'm honest.   A little reading tells me that scala doesn't allow overriding non-abstract `def`s with `var`s.  But it _does_ allow overriding abstract `def`s.  So the other option would be to just remove the default in the base trait, and force instantiations to provide a value.  Thoughts?

Reply:
Interesting, I'm not sure how I didn't try that! I think that is cleaner as well, and everything seems to work when I do it that way. See newest commit.